scijava/scijava-ops/src/main/java/org/scijava/ops/util/OpWrapper.java

```diff
@@ -2,12 +2,16 @@
 
 import java.lang.reflect.ParameterizedType;
 import java.lang.reflect.Type;
+import java.util.UUID;
 
+import org.scijava.ops.OpInfo;
+import org.scijava.ops.hints.Hints;
+import org.scijava.ops.provenance.OpHistory;
 import org.scijava.plugin.SciJavaPlugin;
 import org.scijava.types.Types;
 
 public interface OpWrapper<T> extends SciJavaPlugin {
-	T wrap(T op, Type reifiedType);
+	T wrap(T op, OpInfo info, Hints hints, OpHistory history, UUID uuid, Type reifiedType);
```

Comment:
What we could do is make `RichOp` consist of two things: 1) an `Object op` and 2) `OpBookkeeping stuff`, and `OpBookkeeping` is a container for `OpInfo`, `Hints`, `OpHistory`, `UUID`, and anything else we want to add later.

Reply:
@ctrueden this change is made in 8a87442e.

We now have `RichOp`, `DefaultRichOp`, and `OpMetadata`. Part of me wonders if we actually need all of these types, but I think they are nice.